### DISCUSSION TITLE
renode launch with custom yocto-dev build

### DISCUSSION THREAD

**Discussion by eric-riscv** [polarfire-soc/polarfire-soc-documentation#173] [Linux build systems for PolarFire SoC]
Given launching renode successfully with the default [icicle-kit.resc](https://github.com/renode/renode/blob/master/scripts/single-node/icicle-kit.resc) file, and given a default yocto-dev mpfs-dev-cli build according to [microchip instructions](https://github.com/polarfire-soc/meta-polarfire-soc-yocto-bsp), attempting to replace (only) the fitImage file in the renode launch script with the fitImage file from the yocto-dev build results in:

Verifying Hash Integrity ... sha256 error!
Unsupported hash algorithm for 'hash-1' hash node in 'kernel-1' image node

Is there an "easy" solution to turn off compressing the kernel which is how the original ant micro build appears to have that setup?  Or perhaps you have instructions for completely rebuilding all of the icicle inputs/files to the renode icicle-kit.resc script such that there is a complete custom build from source?  (and pointers/helpers for building yocto for the renode target?)

====

Failed attempt with custom yocto-dev fitImage:

04:09:43.4224 [INFO] mmuart1: [+6.64ms host +0s virt 2.64s virt from start]   Running Linux...
04:09:43.4282 [INFO] mmuart1: [+5.84ms host +0s virt 2.64s virt from start]   ## Loading kernel from FIT Image at 88300000 ...
04:09:43.4402 [INFO] mmuart1: [+12.01ms host +0.6ms virt 2.64s virt from start]      Using 'conf-microchip_mpfs-icicle-kit.dtb' configuration
04:09:43.4496 [INFO] mmuart1: [+9.34ms host +0.3ms virt 2.64s virt from start]      Trying 'kernel-1' kernel subimage
04:09:43.4500 [INFO] mmuart1: [+0.44ms host +0s virt 2.64s virt from start]        Description:  Linux kernel
04:09:43.4511 [INFO] mmuart1: [+1.1ms host +0s virt 2.64s virt from start]        Type:         Kernel Image
04:09:43.4520 [INFO] mmuart1: [+0.86ms host +0s virt 2.64s virt from start]        Compression:  gzip compressed
04:09:43.4598 [INFO] mmuart1: [+7.82ms host +0.3ms virt 2.64s virt from start]        Data Start:   0x883000fc
04:09:43.4676 [INFO] mmuart1: [+7.82ms host +0.3ms virt 2.64s virt from start]        Data Size:    4328359 Bytes = 4.1 MiB
04:09:43.4681 [INFO] mmuart1: [+0.56ms host +0s virt 2.64s virt from start]        Architecture: RISC-V
04:09:43.4689 [INFO] mmuart1: [+0.8ms host +0s virt 2.64s virt from start]        OS:           Linux
04:09:43.4737 [INFO] mmuart1: [+4.74ms host +0.3ms virt 2.64s virt from start]        Load Address: 0x1000200000
04:09:43.4742 [INFO] mmuart1: [+0.54ms host +0s virt 2.64s virt from start]        Entry Point:  0x1000200000
04:09:43.4803 [INFO] mmuart1: [+6.1ms host +0.3ms virt 2.64s virt from start]        Hash algo:    sha256
04:09:43.4852 [INFO] mmuart1: [+4.83ms host +0.3ms virt 2.64s virt from start]        Hash value:   f8af4c4ad5d8817c083dda0f7e230ddc00c790cfab2e05117144307bb577180b
04:09:43.4910 [INFO] mmuart1: [+5.88ms host +0.3ms virt 2.64s virt from start]      Verifying Hash Integrity ... sha256 error!
04:09:43.4914 [INFO] mmuart1: [+0.33ms host +0s virt 2.64s virt from start]   Unsupported hash algorithm for 'hash-1' hash node in 'kernel-1' image node
04:09:43.4917 [INFO] mmuart1: [+0.32ms host +0s virt 2.64s virt from start]   Bad Data Hash
04:09:43.4922 [INFO] mmuart1: [+0.51ms host +0s virt 2.64s virt from start]   ERROR: can't get kernel image!

====

Same section of log with the original https downloaded fitImage from the original icicle-kit.resc file:

04:06:13.4649 [INFO] mmuart1: [+8.34ms host +0s virt 2.64s virt from start]   Running Linux...
04:06:13.4700 [INFO] mmuart1: [+5.12ms host +0s virt 2.64s virt from start]   ## Loading kernel from FIT Image at 88300000 ...
04:06:13.4793 [INFO] mmuart1: [+9.22ms host +0.6ms virt 2.64s virt from start]      Using 'conf@microchip_icicle-kit-es-a000-microchip.dtb' configuration
04:06:13.4841 [INFO] mmuart1: [+4.79ms host +0.3ms virt 2.64s virt from start]      Trying 'kernel@1' kernel subimage
04:06:13.4846 [INFO] mmuart1: [+0.55ms host +0s virt 2.64s virt from start]        Description:  Linux kernel
04:06:13.4864 [INFO] mmuart1: [+1.79ms host +0s virt 2.64s virt from start]        Type:         Kernel Image
04:06:13.4922 [INFO] mmuart1: [+5.76ms host +0.3ms virt 2.64s virt from start]        Compression:  uncompressed
04:06:13.4970 [INFO] mmuart1: [+4.85ms host +0.3ms virt 2.64s virt from start]        Data Start:   0x883000fc
04:06:13.4992 [INFO] mmuart1: [+2.17ms host +0s virt 2.64s virt from start]        Data Size:    6477712 Bytes = 6.2 MiB
04:06:13.4997 [INFO] mmuart1: [+0.55ms host +0s virt 2.64s virt from start]        Architecture: RISC-V
04:06:13.5002 [INFO] mmuart1: [+0.5ms host +0s virt 2.64s virt from start]        OS:           Linux
04:06:13.5099 [INFO] mmuart1: [+9.7ms host +0.3ms virt 2.64s virt from start]        Load Address: 0x80200000
04:06:13.5104 [INFO] mmuart1: [+0.52ms host +0s virt 2.64s virt from start]        Entry Point:  0x80200000
04:06:13.5179 [INFO] mmuart1: [+7.48ms host +0.3ms virt 2.64s virt from start]      Verifying Hash Integrity ... OK
04:06:13.5465 [INFO] mmuart1: [+28.53ms host +1.8ms virt 2.64s virt from start]   ## Loading ramdisk from FIT Image at 88300000 ...
04:06:13.5562 [INFO] mmuart1: [+9.74ms host +0.6ms virt 2.64s virt from start]      Using 'conf@microchip_icicle-kit-es-a000-microchip.dtb' configuration
04:06:13.5605 [INFO] mmuart1: [+4.35ms host +0.3ms virt 2.64s virt from start]      Trying 'ramdisk@initramfs.cpio.gz' ramdisk subimage
04:06:13.5608 [INFO] mmuart1: [+0.25ms host +0s virt 2.64s virt from start]        Description:  buildroot initramfs
04:06:13.5611 [INFO] mmuart1: [+0.32ms host +0s virt 2.64s virt from start]        Type:         RAMDisk Image
04:06:13.5654 [INFO] mmuart1: [+4.27ms host +0.3ms virt 2.65s virt from start]        Compression:  Unknown Compression
04:06:13.5700 [INFO] mmuart1: [+4.65ms host +0.3ms virt 2.65s virt from start]        Data Start:   0x8892d94c
04:06:13.5704 [INFO] mmuart1: [+0.36ms host +0s virt 2.65s virt from start]        Data Size:    10488832 Bytes = 10 MiB
04:06:13.5706 [INFO] mmuart1: [+0.24ms host +0s virt 2.65s virt from start]        Architecture: RISC-V
04:06:13.5783 [INFO] mmuart1: [+7.7ms host +0.3ms virt 2.65s virt from start]        OS:           Linux
04:06:13.5786 [INFO] mmuart1: [+0.29ms host +0s virt 2.65s virt from start]        Load Address: 0x84000000
04:06:13.5790 [INFO] mmuart1: [+0.38ms host +0s virt 2.65s virt from start]        Entry Point:  unavailable
04:06:13.5894 [INFO] mmuart1: [+10.41ms host +0.6ms virt 2.65s virt from start]      Verifying Hash Integrity ... OK
04:06:13.5985 [INFO] mmuart1: [+9.04ms host +0.6ms virt 2.65s virt from start]      Loading ramdisk from 0x8892d94c to 0x84000000

**Comment by hughbreslin** [comment]
Hey @eric-riscv just to mention we are looking at this and will hopefully have a reasonable answer in a few days.

**Reply by Micko8** [reply]
Hello,
I'm facing the same problem. Cannot use my custom yocto image with Renode. Has this issue been solved already ?
Cheers
